app/views/conference/_all_events.html.erb

```diff
@@ -1,19 +1,73 @@
 <h1 class="text-center">Program for <%= @conference.title %></h1>
-<% @conference.program.events.confirmed.each do |event| %>
-    <div>
-        <h3>
-          <%= link_to event.title, conference_program_proposal_path(@conference.short_title, event.id) %>
-          <br>
-          <small>
-            <%= event.subtitle %>
-          </small>
-        </h3>
-        <h4>
-          presented by <%= event.speaker_names %>
-        </h4>
-        <p>
-          <%= truncate(event.abstract, :length => 400) -%>
-          <%= link_to 'more', conference_program_proposal_path(@conference.short_title, @conference.program.id, event.id) if event.abstract.length > 400 %>
-        </p>
+<div class="dropdown program-dropdown">
+  <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown">
+    Dates
+    <span class="caret"></span>
+  </button>
+  <ul class="dropdown-menu">
+    <% @dates.each do |date| %>
+      <li class='li-dropdown-program'>
+        <%= link_to date, "#program-" + "#{date}", class: "program-selector#{ ' no-events-day' unless any_event_date?(@scheduled_events, date) }" %>
+      </li>
+    <% end %>
+    <% if @unscheduled_events.any? %>
+      <li class='li-dropdown-program'>
+        <%= link_to('Unscheduled', "#program-unscheduled", class: 'program-selector') %>
+      </li>
+    <% end %>
+  </ul>
+</div>
+
+<div class="row">
+
+  <!-- scheduled events -->
+  <% date = nil %>
+  <% time = nil %>
+  <% @scheduled_events.each do |event| %>
+    <% unless event.date_string.eql?(date) %>
+      <div class='col-xs-12 col-md-12 all-events-title'>
+        <span class='date-title' id='program-<%= event.date_string %>'>
+          <%= date = event.date_string %>
+        </span>
+        <a title="Go up" class="pull-right" href="#program">
+          <i class="fa fa-angle-double-up fa-lg" aria-hidden="true"></i>
+        </a>
+      </div>
+    <% end %>
+    <% if !event.time_string.eql?(time) %>
+      <div class='col-xs-12 col-md-1 start-time'>
```

Comment:
Please do not mix bootstrap col's with other CSS especially if you manipulate the [box-model](https://www.w3.org/TR/css3-box/) as this can break the grid